File: python/sdk/distortion_correct_demo.py
# -*- coding:utf-8 -*-
from ais_sdk.gettoken import get_token
from ais_sdk.utils import encode_to_base64
from ais_sdk.utils import decode_to_wave_file
from ais_sdk.distortion_correct import distortion_correct
from ais_sdk.distortion_correct import distortion_correct_aksk
import json

if __name__ == '__main__':
    #
    # access moderation distortion correct.post data by token
    #
    user_name = '******'
    password = '******'
    account_name = '******'  # the same as user_name in commonly use

    demo_data_url = 'https://ais-sample-data.obs.myhwclouds.com/tagging-normal.jpg'
    token = get_token(user_name, password, account_name)

    #call interface use the url correction is true means do not correction
    result = distortion_correct(token, "", demo_data_url, True)
    result_obj = json.loads(result)
    decode_to_wave_file(result_obj['result']['data'], 'data/modeation_distortion-correct_first.png')
    print result

    # call interface use the file
    result = distortion_correct(token, encode_to_base64('data/modeation_distortion.jpg'), '', True)
    result_obj = json.loads(result)
    decode_to_wave_file(result_obj['result']['data'], 'data/modeation_distortion-correct_second.png')
    print result

    #
    # access moderation distortion correct.post data by ak,sk
    app_key = "*************"
    app_secret = "************"

    demo_data_url = 'https://ais-sample-data.obs.myhwclouds.com/tagging-normal.jpg'
    #call interface use the url correction is true means do not correction
    result = distortion_correct_aksk(app_key, app_secret, "", demo_data_url, True)
    result_obj = json.loads(result)
    decode_to_wave_file(result_obj['result']['data'], 'data/modeation_distortion-third.png')
    print result

    # call interface use the file
    result = distortion_correct_aksk(app_key, app_secret, encode_to_base64('data/modeation_distortion.jpg'), '', True)
    print result

File: javascript/distortion_correct_demo.js
/**
 * 扭曲校正服务的使用示例
 */
var discor = require("./ais_sdk/distortion_correct");
var token = require("./ais_sdk/gettoken");
var utils = require("./ais_sdk/utils");

var username = "*******";       // 配置用户名
var domainname = "*******";     // 配置用户名
var password = "*******";       // 密码
var regionName = "cn-north-1";  // 配置区域信息

var filepath = "./data/modeation_distortion.jpg";
var data = utils.changeFileToBase64(filepath);

/**
 * token 方式获取结果
 * @type {string}
 */
demo_data_url = "https://ais-sample-data.obs.myhwclouds.com/tagging-normal.jpg";
token.getToken(username, domainname, password, regionName, function (token) {

    discor.distortion_correct(token, data, "", true, function (result) {
        var resultObj = JSON.parse(result);

        if (resultObj.result.data !== "") {
            utils.getFileByBase64Str("./data/modeation_distortion-1.jpg", resultObj.result.data);
        }else{
            console.log(result);
        }
    });

    discor.distortion_correct(token, "", demo_data_url, true, function (result) {
        var resultObj = JSON.parse(result);

        if (resultObj.result.data !== "") {
            utils.getFileByBase64Str("./data/modeation_distortion-2.jpg", resultObj.result.data);
        }else{
            console.log(result);
        }
    });
});

/**
 * aksk 方式获取结果
 * @type {string}
 */
var app_key = "*************";
var app_secret = "************";

discor.distortion_correct_aksk(app_key, app_secret, data, "", true, function (result) {
    var resultObj = JSON.parse(result);

    if (resultObj.result.data !== "") {
        utils.getFileByBase64Str("./data/modeation_distortion-2.jpg", resultObj.result.data);
    }else{
        console.log(result);
    }
});

discor.distortion_correct_aksk(app_key, app_secret, "", demo_data_url, true, function (result) {
    var resultObj = JSON.parse(result);

    if (resultObj.result.data !== "") {
        utils.getFileByBase64Str("./data/modeation_distortion-2.jpg", resultObj.result.data);
    }else{
        console.log(result);
    }
});